### PR TITLE
Slight logic adjustments to reduce reading in backend

### PR DESCRIPTION
**Pull request by joesho112358**
Noticed some inconsistencies while reading through, specifically:
```
if x:
  return y
else:
  return z
```

vs

```
if x:
  return y
return z
```
I went with the later because it is a little less typing/reading. Open to going the other way, too, but figured it should be consistent in the file.

### CODE REVIEW

File: keras/backend.py [hunk 1]
```diff
@@ -6117,10 +6089,10 @@ def conv2d_transpose(
         raise ValueError("Unknown data_format: " + str(data_format))
 
     # `atrous_conv2d_transpose` only supports NHWC format, even on GPU.
-    if data_format == "channels_first" and dilation_rate != (1, 1):
-        force_transpose = True
-    else:
-        force_transpose = False
+    force_transpose = data_format == "channels_first" and dilation_rate != (
```
Comment (qlzh727) [inline keras/backend.py:6092]: The format here is quite weird, how about wrap the both cause in a () and break at the `and`

**Review by qlzh727 — CHANGES_REQUESTED** [review]
Thanks for the clean up, just a minor comment about one spot.

File: keras/backend.py (hunk 1 above)
Comment (joesho112358) [reply keras/backend.py:6092]: oh yeah! i saw that after running `sh shell/format.sh` and was surprised it made it like that

Comment (joesho112358) [reply keras/backend.py]: updated

**Comment by joesho112358** [comment]
i see no changes when running locally:
```
$ sh shell/format.sh
All done! ✨ 🍰 ✨
727 files left unchanged.
```
i am not sure how to handle this discrepancy.

File: keras/backend.py [hunk 2]
```diff
@@ -409,18 +408,16 @@ def _internal_get_learning_phase(graph):
...
-            return tf.compat.v1.placeholder_with_default(
-                False, shape=(), name="keras_learning_phase"
-            )
+    with name_scope(""):
+        return tf.compat.v1.placeholder_with_default(
+            False, shape=(), name="keras_learning_phase"
+        )
 
 
 @keras_export("keras.backend.set_learning_phase")
```
Comment (fchollet) [inline keras/backend.py:423]: Please revert changes in this function -- it would only run in V1 graph mode, which isn't tested on GitHub CI.

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

**Comment by fchollet** [comment]
To be clear, I meant changes in `set_learning_phase` -- not https://github.com/keras-team/keras/pull/17537/commits/7efadea511671a047af8d63a2a7936c6abbd3b19

**Comment by joesho112358** [comment]
> To be clear, I meant changes in `set_learning_phase` -- not [7efadea](https://github.com/keras-team/keras/commit/7efadea511671a047af8d63a2a7936c6abbd3b19)

Hi! You're going to have to clarify further. From what I can tell, I neither touched `set_learning_phase` nor any of the functions called in it. Specifically, `set_learning_phase` calls `deprecated_internal_set_learning_phase` which calls `_internal_set_learning_phase`, and I don't see where I touched any of these functions.

**Comment by fchollet** [comment]
Sorry, I got confused by code truncation in the code review interface. I meant `_get_session()`.

**Review by fchollet — COMMENTED** [review]
Please fix the lint errors.

**Comment by joesho112358** [comment]
@fchollet [This commit](https://github.com/keras-team/keras/pull/17537/commits/ba834a979f83531d9136a274ce1eb558b6531d12) is from the discussion above with @qlzh727 and it looks to be one of the reasons for the lint error. Should I just run the `sh shell/format.sh` script again to put it back the way it was?

**Comment by joesho112358** [comment]
@fchollet The lint issue is fixed, but please look up at @qlzh727 's comment about this part not looking quite right:
https://github.com/keras-team/keras/pull/17537/commits/d8fcd9bc0156d8634d95adf5ec3a12191a6e2360

**Comment by fchollet** [comment]
> please look up at @qlzh727 's comment about this part not looking quite right:
https://github.com/keras-team/keras/commit/d8fcd9bc0156d8634d95adf5ec3a12191a6e2360

Thanks -- no concern on my side, this is just an ugly autoformatter edge case. We could rephrase the code to fix it, but it's not super important.

**Review by fchollet — APPROVED** [review]
LGTM

**Review by tekeuange23 — APPROVED** [review]

**Comment by fchollet** [comment]
We seem to have a few test failures here -- please take a look at the logs.

**Comment by tekeuange23** [comment]
I can look at it maybe if @joesho112358 is unavailable

**Comment by joesho112358** [comment]
@tekeuange23 I'm fine with that if you want to take a crack at it. I might not get to it until the weekend and maybe not even then

**Comment by gbaned** [comment]
Hi @tekeuange23 Any update on this PR? Please. Thank you!

**Comment by tekeuange23** [comment]
hi @gbaned!!! 

sorry for the delayed response. I have some issues with my setup cause I am a new contributor on this project. not sure to fix that soon to properly code and run the tests. 

Sorry @joesho112358.

**Comment by joesho112358** [comment]
@tekeuange23 no worries

I will see what I can do in a bit

**Pull request closed without merging**